core-library/src/main/java/org/silverpeas/core/index/search/model/QueryDescription.java

```diff
@@ -100,6 +103,14 @@ public QueryDescription(String query) {
    */
   public void setQuery(String query) {
     this.query = (query == null) ? "" : query.toLowerCase();
+    // string already in lower case, means "and" "And" "AND" will works.
+    this.query = findAndReplace(this.query, " and ", " AND ");
+    this.query = findAndReplace(this.query, " or ", " OR ");
+    this.query = findAndReplace(this.query, " not ", " NOT ");
+    if (this.query.indexOf("not ") == 0) {
+      this.query = "NOT " + this.query.substring(4, this.query.length());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)